.gitignore

```diff
@@ -49,22 +49,22 @@ perl/Makefile.config
 /src/libexpr/parser-tab.output
 /src/libexpr/nix.tbl
 /src/libexpr/tests
-/tests/unit/libexpr/libnixexpr-tests
+/src/libexpr-test/libnixexpr-tests
 
 # /src/libfetchers
-/tests/unit/libfetchers/libnixfetchers-tests
+/src/libfetchers-test/libnixfetchers-tests
 
 # /src/libflake
-/tests/unit/libflake/libnixflake-tests
+/src/libflake-test/libnixflake-tests
 
 # /src/libstore/
 *.gen.*
 /src/libstore/tests
-/tests/unit/libstore/libnixstore-tests
+/src/libstore-test/libnixstore-tests
 
 # /src/libutil/
 /src/libutil/tests
-/tests/unit/libutil/libnixutil-tests
```

Comment:
Are these worth moving? Symlinks seemed to work fine for the time being.
If we're moving them, should we rename them to match their meson names, e.g. `src/nix-util-test`?

Reply:
I think so: as soon as I remove things from the old build system, I want to clean up the code organization.

> If we're moving them, should we rename them to match their meson names, e.g. `src/nix-util-test`?

Oh yes!

Reply:
I have a bad feeling about these file moves.

We should make it easier, not harder to finish existing work.
Also, IIRC we didn't agree to insert this into our short term plans until we saw that the impact would be manageable, because no moves/renames were required. (Not sure I recall the order of events correctly, and we probably need better notes for this kind of thing, so grain of salt, but it is how the reasoning works for me at least)

The symlink solution I used for the `libutil` tests was trivially without conflicts (with existing PRs) and worked fine without having to move anything.
I do want these to be renamed in the medium term, but I think we should at least try to find somewhat of a good time for this, after finishing at least _some_ other work and before starting whole-new improvement projects.